Project/Assets/ML-Agents/Examples/SharedAssets/Scripts/ModelOverrider.cs

```diff
@@ -32,27 +45,56 @@ void GetAssetPathFromCommandLine()
         {
             m_BehaviorNameOverrides.Clear();
 
+            var maxEpisodes = 0;
+
             var args = Environment.GetCommandLineArgs();
-            for (var i = 0; i < args.Length-2; i++)
+            for (var i = 0; i < args.Length - 1; i++)
             {
-                if (args[i] == k_CommandLineFlag)
+                if (args[i] == k_CommandLineModelOverrideFlag && i < args.Length-2)
                 {
                     var key = args[i + 1].Trim();
                     var value = args[i + 2].Trim();
                     m_BehaviorNameOverrides[key] = value;
                 }
+                else if (args[i] == k_CommandLineQuitAfterEpisodesFlag)
+                {
+                    Int32.TryParse(args[i + 1], out maxEpisodes);
+                }
+            }
+
+            if (m_BehaviorNameOverrides.Count > 0)
+            {
+                // If overriding models, set maxEpisodes to 1 or the command line value
+                m_MaxEpisodes = maxEpisodes > 0 ? maxEpisodes : 1;
+                Debug.Log($"setting m_MaxEpisodes to {maxEpisodes}");
             }
         }
 
         void OnEnable()
         {
+            m_Agent = GetComponent<Agent>();
+
             GetAssetPathFromCommandLine();
             if (m_BehaviorNameOverrides.Count > 0)
             {
                 OverrideModel();
             }
         }
 
+        void FixedUpdate()
+        {
+            if (m_MaxEpisodes > 0)
+            {
+                if (m_NumSteps > m_MaxEpisodes * m_Agent.maxStep)
```

Comment:
I want to make sure we run at least `Agent.maxSteps` steps, so even if an Agent finishes early we keep running at least a full episode for the rest. This approach made the most sense to me, so that if maxSteps changes in an example, we don't need to change any other config. Open to other ideas though.